app/src/main/java/campus/tech/kakao/map/MainActivity.kt

```diff
@@ -82,11 +83,23 @@ class MainActivity : AppCompatActivity() {
     fun setupRecyclerViews() {
         searchAdapter = SearchAdapter { result ->
             viewModel.addSearch(result.place_name)
+            val intent = Intent(this, MapActivity::class.java).apply {
+                putExtra("place_name", result.place_name)
+                putExtra("place_address", result.address_name)
+                putExtra("place_x", result.x)
+                putExtra("place_y", result.y)
+            }
+            startActivity(intent)
         }
 
-        savedSearchAdapter = SavedSearchAdapter { place ->
-            viewModel.removeSearch(place)
-        }
+        savedSearchAdapter = SavedSearchAdapter (
+            onSearchClicked = { searchQuery ->
```

Comment:
nit; 아래처럼 작성도 가능합니다.
```suggestion
            onSearchClicked = viewModel::searchSavedPlace
```